loki/transformations/transpile/fortran_c.py

```diff
@@ -157,18 +182,55 @@ def transform_subroutine(self, routine, **kwargs):
+            bind_name = None if self.language == 'c' else f'{routine.name.lower()}_c_launch'
+            wrapper = self.generate_iso_c_wrapper_routine(routine, self.c_structs, bind_name=bind_name)
             contains = Section(body=(Intrinsic('CONTAINS'), wrapper))
             self.wrapperpath = (path/wrapper.name.lower()).with_suffix('.F90')
             module = Module(name=f'{wrapper.name.upper()}_MOD', contains=contains)
+            module.spec = Section(body=(Import(module='iso_c_binding'),))
+
+            # Generate C source file from Loki IR
+            c_kernel = self.generate_c_kernel(routine, targets=targets)
+            self.c_path = (path/c_kernel.name.lower()).with_suffix('.c')
             Sourcefile.to_file(source=fgen(module), path=self.wrapperpath)
 
             # Generate C source file from Loki IR
-            c_kernel = self.generate_c_kernel(routine)
+            # c_kernel.spec.prepend(Import(module=f'{c_kernel.name.lower()}.h', c_import=True))
+            for successor in successors:
+                if self.language == 'c':
+                    c_kernel.spec.prepend(Import(module=f'{successor.routine.name.lower()}_c.h', c_import=True))
+                else:
+                    # TODO: should include .h file, however problem compiling/running multiple compilation units ...
+                    if not isinstance(successor, ProcedureItem):
+                        c_kernel.spec.prepend(Import(module=f'{successor.routine.name.lower()}_c.c', c_import=True))
+
+            if depth == 1:
+                if self.language != 'c':
+                    c_kernel_launch = c_kernel.clone(name=f"{c_kernel.name}_launch", prefix="extern_c")
+                    self.generate_c_kernel_launch(c_kernel_launch, c_kernel)
+                    self.c_path = (path/c_kernel_launch.name.lower()).with_suffix('.h')
+                    Sourcefile.to_file(source=self.codegen(c_kernel_launch, extern=True), path=self.c_path)
+
+            assignments = FindNodes(Assignment).visit(c_kernel.body)
+            assignments2remove = ['griddim', 'blockdim']
+            assignment_map = {assignment: None for assignment in assignments
+                    if assignment.lhs.name.lower() in assignments2remove}
+            # == block_dim.index.lower() or assignment.lhs.name.lower() in [_.lower() for _ in horizontal.bounds]}
+            c_kernel.body = Transformer(assignment_map).visit(c_kernel.body)
+
+            if depth > 1:
+                c_kernel.spec.prepend(Import(module=f'{c_kernel.name.lower()}.h', c_import=True))
+            self.c_path = (path/c_kernel.name.lower()).with_suffix('.c')
+            Sourcefile.to_file(source=self.codegen(c_kernel), path=self.c_path)
+            self.c_path = (path/c_kernel.name.lower()).with_suffix('.h')
+            Sourcefile.to_file(source=self.codegen(c_kernel, header=True), path=self.c_path)
             self.c_path = (path/c_kernel.name.lower()).with_suffix('.c')
```

Comment:
Wouldn't it make more sense to not overwrite the `c_path` property and just specify the header path as a separate variable?
```suggestion
            header_path = (path/c_kernel.name.lower()).with_suffix('.h')
            Sourcefile.to_file(source=self.codegen(c_kernel, header=True), path=header_path)
```